crates/miden-lib/src/account/auth/mod.rs

```diff
@@ -39,3 +42,69 @@ impl From<RpoFalcon512> for AccountComponent {
         .with_supports_all_types()
     }
 }
+
+/// An [`AccountComponent`] implementing a conditional RpoFalcon512 signature scheme for
+/// authentication of transactions.
+///
+/// This component only requires authentication when any of the specified procedures are called
+/// during the transaction. It stores a list of procedure roots that require authentication, and
+/// the signature verification is only performed if at least one of these procedures was invoked.
+///
+/// It exports the procedure `auth__tx_rpo_falcon512_conditional`, which:
+/// - Checks if any of the specified procedures were called during the transaction
+/// - If none were called, authentication is skipped
+/// - If at least one was called, performs standard RpoFalcon512 signature verification
+///
+/// The storage layout is:
+/// - Slot 0(value): Public key (same as RpoFalcon512)
+/// - Slot 1(value): Number of tracked procedures
+/// - Slot 2(map): A map with tracked procedure roots
+///
+/// This component supports all account types.
+pub struct RpoFalcon512ProcedureACL {
+    public_key: PublicKey,
+    trigger_procedures: Vec<Digest>,
+}
+
+impl RpoFalcon512ProcedureACL {
+    /// Creates a new [`RpoFalcon512ProcedureACL`] component with the given `public_key` and
+    /// list of procedure roots that require authentication.
+    ///
+    /// # Panics
+    /// Panics if more than 253 procedures are tracked (to leave room for the public key and count).
+    pub fn new(public_key: PublicKey, trigger_procedures: Vec<Digest>) -> Self {
+        assert!(
+            trigger_procedures.len() <= u8::MAX as usize - 2,
+            "Cannot track more than 253 procedures"
+        );
```

Comment:
~~I would prefer returning an error instead of panicking. (We do this in some parts of the codebase, but imo we should refactor user-triggerable panics to errors. Imo, invariant violations can panic, but that also depends a bit on the details. For instance, `AccountComponent::new` is generally fine because we can assert correctness on a for every component we write.)~~

~~We may not want to add a dedicated `RpoFalcon512Error` type, so we could just return `AccountError` and use `AccountError::AssumptionViolated`. (Though if you use it, I think we should refactor it to be like `TransactionProverError::Other` so we can include a source error for other use cases).~~

Edit: Actually, this assertion can just be removed now that the procedure roots are stored in a storage map, right?

Reply:
The assertion is definitely outdated 👍🏼 

However, maybe we can rework it, since technically we only allow 256 procedures on the account interface. I think it's ok if the auth component is aware of this limitation (`AccountCode::MAX_NUM_PROCEDURES`). WDYT?

Reply:
Oh, yeah, makes sense! I hadn't considered the upper limit to the number of account procedures. In that case, I agree let's keep it and use `AccountCode::MAX_NUM_PROCEDURES` in the calculation to make it clear that this is where the limit comes from.